File: 10678 The Grazing Cow/solution.c
#include <stdio.h>
#include <math.h>

int main() {
    double PI = acos(-1);
    int N;
    scanf("%d", &N);
    while(N--) {
        double D, L;
        scanf("%lf %lf", &D, &L);
        printf("%.3f\n", L * sqrt(L * L - D * D) * PI / 4);
    }
    return 0;
}

File: 10678 The Grazing Cow/solution.java
import java.io.*;

public class Main {
	public static void main (String[] args) throws IOException {
		StreamTokenizer st = new StreamTokenizer(System.in);
		st.nextToken();
		int N = (int)st.nval;
		StringBuilder output = new StringBuilder();
		while(N-- > 0) {
		    st.nextToken();
		    double D = st.nval;
		    st.nextToken();
		    double L = st.nval;
		    output.append(String.format("%.3f\n", L * Math.sqrt(L * L - D * D) * Math.PI / 4));
		}
		System.out.print(output);
	}
}
